api/app/graphql/types/mutation_type.rb

```diff
@@ -1,12 +1,37 @@
 # frozen_string_literal: true
 
+require_all 'lib'
+
 module Types
   class MutationType < Types::BaseObject
+    include TokenAuthorization
     # TODO: remove me
-    field :test_field, String, null: false,
-                               description: 'An example field added by the generator'
-    def test_field
-      'Hello World'
+    field :register, Types::AppResponseType, null: false, description: "Register a new user" do
+      argument :phone, String, required: true
+      argument  :email, String, required:false
+      argument :date_of_birth, GraphQL::Types::ISO8601DateTime, required: true
+      argument :isMuslim, Boolean, required: false
+      argument :password, String, required: true
+    end
+  
+    def register(**kwargs)
+    
+      user = User.new(kwargs)
+
+      if user.save
+        authToken = TokenAuthorization.encode( { user_id: user.id } )
+        {
+          status: 'success',
+          message: 'user registered successfully',
+          body: authToken
+        }
+      else
+        {
+          status: 'unprocessible_entity',
+          message: 'user registration failed',
+          body: user.errors.full_messages.join(', ')
```

Comment:
After changing the type, you will need to return the array of errors instead of joining all messages into a single string. This will make things more readable.
Eg.

```ruby
{
  body: user.errors
}
```